tests/src/test/java/org/eclipse/hono/tests/coap/CoapTestBase.java

```diff
@@ -1311,6 +1318,141 @@ public void testUploadMessagesWithTtdThatReplyWithOneWayCommand(
                 });
     }
 
+    /**
+     * Ensures that a timestamp is within 2 seconds of current system clock.
+     * This is necessary because the test containers may have a clock drift of a few milliseconds
+     * from the host clock.
+     *
+     * @param ts the timestamp to check
+     */
+    private void assertTimestampWithinRangeOfCurrentTime(final long ts) {
```

Comment:
FMPOV this is not something that the adapter can guarantee as it mainly depends on the networking infrastructure (UDP packets may get dropped by intermediary routers at any time).

Reply:
True. The problem was that the integration tests run in multiple containers and the system time in them seems to differ by a few milliseconds.
For example, I was getting failed tests where the time value in the option received from the adapter was a few milliseconds into the future.

So what would you suggest doing here? I can forego checking the value altogether of course and just check that there _is_ a long value in the response option, but that seems a bit off to me.

The idea with this buffer was not to provide any guarantee that the feature works within a specific time but simply to provide some buffer so the unit tests don't fail. There is an attempt to explain this in the JavaDoc for this method but I could make it more explicit. But I'm also very open to other solutions to this.

Reply:
Ok, I see your point. IMHO you have already made sure that the implemetation is using the current system time by means of the unit tests. So FMPOV we could indeed simply check if the response actually contains a long valued TimeOption here ...

Reply:
okay, that sounds good. I'll change that

Reply:
done.